### PR TITLE
Build: Improve `npm run perf` command (fixes #5028)

### LINKED ISSUE #5028: improve the `npm run perf` command
Currently, the `npm run perf` command is running tests for performance of single file.
I would like to add tests for performance of multi files.
Because the characteristics of performance can be different between those two. (https://github.com/eslint/eslint/pull/4428#issuecomment-156789289)

That result is something like:

```
$ npm run perf

> eslint@2.0.0-beta.1 perf C:\Users\t-nagashima.AD\Documents\GitHub\eslint
> node Makefile.js perf

Single File:
  CPU Speed is 3392 with multiplier 7500000
  Performance Run #1:  2422.559916ms
  Performance Run #2:  2488.714065ms
  Performance Run #3:  2452.773965ms
  Performance Run #4:  2413.956653ms
  Performance Run #5:  2392.814938ms
  Performance budget exceeded: 2422.559916ms (limit: 2211.0849056603774ms)

Multi Files:
  CPU Speed is 3392 with multiplier 24000000
  Performance Run #1:  7571.863146ms
  Performance Run #2:  7449.154811ms
  Performance Run #3:  7428.685312ms
  Performance Run #4:  7480.038359ms
  Performance Run #5:  7559.607224ms
  Performance budget exceeded: 7480.038359ms (limit: 7075.471698113208ms)


Load performance Run #1:  191.676006ms
Load performance Run #2:  193.562554ms
Load performance Run #3:  193.574326ms
Load performance Run #4:  199.532498ms
Load performance Run #5:  192.297209ms

Load Performance median :  193.562554ms
```

I'm happy to make a PR if accepted.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/30113170-improve-the-npm-run-perf-command?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by mysticatea**
Fixes #5028.

- A performance test for multi files was added. 
  This test is using `eslint@1.10.3` as the test target.
  This repository has 450 js files in `lib` and `tests/lib` :)
- `.eslintrc` of performance tests came to turn all rules on.

This is new output of this PR:

```
> eslint@2.0.0-beta.2 perf C:\Users\t-nagashima.AD\Documents\GitHub\eslint
> node Makefile.js perf

Loading:
  Load performance Run #1:  193.219655ms
  Load performance Run #2:  193.38688ms
  Load performance Run #3:  191.784973ms
  Load performance Run #4:  191.239836ms
  Load performance Run #5:  203.159501ms

  Load Performance median:  193.219655ms


Single File:
  CPU Speed is 3392 with multiplier 13000000
  Performance Run #1:  3749.119887ms
  Performance Run #2:  3758.879529ms
  Performance Run #3:  3745.368525ms
  Performance Run #4:  3815.086891ms
  Performance Run #5:  3864.7341969999998ms

  Performance budget ok:  3758.879529ms (limit: 3832.5471698113206ms)


Multi Files (450 files):
  CPU Speed is 3392 with multiplier 39000000
  Performance Run #1:  9994.710723ms
  Performance Run #2:  9995.715573ms
  Performance Run #3:  10048.651208ms
  Performance Run #4:  10047.002517ms
  Performance Run #5:  9964.838064ms

  Performance budget ok:  9995.715573ms (limit: 11497.641509433963ms)
```

The loading performance test is the fastest, so I moved it to the first.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,3 +12,5 @@ versions.json
 .eslintcache
 .cache
 **/node_modules
+/tests/performance/eslintrc.yml
+/tests/performance/eslint/
```
Comment (gyandeeps) [inline .gitignore]: Why is this ignore necessary?

File: .gitignore [hunk 2]
```diff
@@ -12,3 +12,5 @@ versions.json
 .eslintcache
 .cache
 **/node_modules
+/tests/performance/eslintrc.yml
```
Comment (gyandeeps) [inline .gitignore]: I dont see this file added to this commit?

File: .gitignore (hunk 1 above)
Comment (mysticatea) [inline .gitignore]: Those files are created in `npm run perf` command.
So this is to avoid to make extra diff after the command.

Comment (gyandeeps) [inline .gitignore]: Oh yeah, Sorry I was thinking other way around.

File: Makefile.js [hunk 3]
```diff
@@ -1063,6 +1064,43 @@ target.checkGitCommit = function() {
 };
 
 /**
+ * Downloads a repository which has many js files to test performance with multi files.
+ * Here, it's eslint@1.10.3 (450 files)
+ * @param {function} cb - A callback function.
+ * @returns {void}
+ */
+function downloadMultifilesTestTarget(cb) {
```
Comment (nzakas) [inline Makefile.js]: Can this checkout be done to a temp directory instead? Would be nice to keep it outside the project to avoid any potential issues.

Comment (mysticatea) [inline Makefile.js]: Make sense. I had forgotten the temp directory exists.

Comment (mysticatea) [inline Makefile.js]: I updated this PR with using OS's temporary directory.

**Comment by nzakas** [comment]
Lgtm

**Pull request merged**